libs/market-list/src/lib/use-markets.ts

```diff
@@ -0,0 +1,110 @@
+import { useYesterday } from '@vegaprotocol/react-helpers';
+import * as Schema from '@vegaprotocol/types';
+import isEqual from 'lodash/isEqual';
+import { useEffect, useMemo } from 'react';
+import type {
+  MarketHookDataFragment,
+  MarketHookQuery,
+  MarketsDataHookSubscription,
+  MarketsDataHookSubscriptionVariables,
+} from './__generated__/MarketsHooks';
+import {
+  MarketsDataHookDocument,
+  useMarketsHookQuery,
+} from './__generated__/MarketsHooks';
+
+export const useMarkets = () => {
+  const yesterday = useYesterday();
+  const { data, loading, error, subscribeToMore } = useMarketsHookQuery({
+    variables: {
+      since: new Date(yesterday).toISOString(),
+      interval: Schema.Interval.INTERVAL_I1H,
+    },
+  });
+
+  const ids = useMemo(() => {
+    if (!data?.marketsConnection?.edges.length) return [];
+    return data.marketsConnection.edges.map((e) => e.node.id);
+  }, [data]);
+
+  useEffect(() => {
+    const unsub = subscribeToMore<
+      MarketsDataHookSubscription,
+      MarketsDataHookSubscriptionVariables
+    >({
+      document: MarketsDataHookDocument,
+      variables: { marketIds: ids },
```

Comment:
In this place we are subscribing to market data (including mark price) for all existing markets.
I'm not sure if backend will be happy about it. For sure it will update multiple times per second and it will rerender <TradeGrid> in same frequency

Reply:
TradeGrid wont't be re-rendered here. This hook is only invoked when the market selector is opened and should only affect child components of SelectMarket.

I agree that it might not be great for the backend. But how else are we supposed to render a list of markets that receives updates via subscription? I noticed that the market list on /markets/all doesn't update at all, which is why I started experimenting with a different hooks/apollo based approach.

Reply:
Additionally the marketsData subscription was recently flattened to improve perf for backend, so I think we should use it in parts of the app where we are rendering lists of markets

Reply:
To get live update of market data for every row in market selector only one line of the code needs to be changed.
https://github.com/vegaprotocol/frontend-monorepo/blob/develop/apps/trading/components/select-market/select-market-columns.tsx#L192
`const noUpdate = false`